app/services/nytimes_service.rb

```diff
@@ -7,6 +7,34 @@ def get_obituaries
       req.params['fq'] = 'news_desk:("obituaries")'
       req.params['fq'] = 'section_name:("obituaries")'
     end
+
+    JSON.parse(response.body, symbolize_names: true)
+  end
+
+  def get_obituaries_by_name(name)
+    response = conn.get do |req|
+      req.params['fq'] = "headline:(#{name}) AND section_name:(obituaries)"
+      req.params['sort'] = 'relevance'
+    end
+    JSON.parse(response.body, symbolize_names: true)
+  end
+
+  def get_covid_obituaries
+    response = conn.get do |req|
+      req.params['fq'] = 'section_name:(obituaries)'
+      req.params['q'] = 'coronavirus'
+    end
+    JSON.parse(response.body, symbolize_names: true)
+  end
+
+  def get_obituaries_by_date(beginning_date, ending_date)
+    response = conn.get do |req|
+      req.params['fq'] = "section_name:(obituaries)"
+      req.params['facet_fields'] = 'pub_year'
+      req.params['facet'] = 'true'
+      req.params['begin_date'] = "#{beginning_date}"
+      req.params['ending_date'] = "#{ending_date}"
+    end
```

Comment:
we might have to do some finagling of the date somewhere along the way. We are (probably) going to pass in a 4 digit year and in order to get all obits for the year the beginning and end date will need to be formatting yyyy0101 and yyyy1231. If we want to have search by specific date or month, we will just need to format the date accordingly for the NYT api to be happy.